File: src/main/java/com/anur/config/LogConfigHelper.java
package com.anur.config;

import java.util.function.Function;

/**
 * Created by Anur IjuoKaruKas on 2019/3/4
 */
public class LogConfigHelper extends ConfigHelper {

    private static String relativelyPath;

    public static String getBaseDir() {
        relativelyPath = getLogBasePath();
        if (relativelyPath == null) {// 并发没关系，这个并不会变动
            relativelyPath = System.getProperty("user.dir");
        }
        return relativelyPath + "\\" + InetSocketAddressConfigHelper.getServerName();
    }

    public static String getLogBasePath() {
        return getConfig(ConfigEnum.LOG_BASE_PATH, Function.identity());
    }

    public static int getIndexInterval() {
        return getConfig(ConfigEnum.LOG_INDEX_INTERVAL, Integer::valueOf);
    }

    public static int getMaxIndexSize() {
        return getConfig(ConfigEnum.LOG_MAX_INDEX_SIZE, Integer::valueOf);
    }

    public static int getMaxLogMessageSize() {
        return getConfig(ConfigEnum.LOG_MAX_MESSAGE_SIZE, Integer::valueOf);
    }

    public static int getMaxLogSegmentSize() {
        return getConfig(ConfigEnum.LOG_MAX_SEGMENT_SIZE, Integer::valueOf);
    }

    public static void main(String[] args) {
        System.out.println(System.getProperty("user.dir"));
    }
}

File: src/main/java/com/anur/config/LogConfigHelper.kt
package com.anur.config

import com.anur.config.common.ConfigHelper
import com.anur.config.common.ConfigurationEnum

/**
 * Created by Anur IjuoKaruKas on 2019/7/5
 *
 * 日志配置相关读取类
 */
class LogConfigHelper {

    companion object : ConfigHelper() {
        private var relativelyPath: String = System.getProperty("user.dir")

        fun getBaseDir(): String {
            return relativelyPath + "/" + InetSocketAddressConfigHelper.getServerName()
        }

        fun getIndexInterval(): Int {
            return getConfig(ConfigurationEnum.LOG_INDEX_INTERVAL) { Integer.valueOf(it) } as Int
        }

        fun getMaxIndexSize(): Int {
            return getConfig(ConfigurationEnum.LOG_MAX_INDEX_SIZE) { Integer.valueOf(it) } as Int
        }

        fun getMaxLogMessageSize(): Int {
            return getConfig(ConfigurationEnum.LOG_MAX_MESSAGE_SIZE) { Integer.valueOf(it) } as Int
        }

        fun getMaxLogSegmentSize(): Int {
            return getConfig(ConfigurationEnum.LOG_MAX_SEGMENT_SIZE) { Integer.valueOf(it) } as Int
        }
    }
}